lib/net/ldap/password.rb

```diff
@@ -6,33 +6,29 @@
 
 class Net::LDAP::Password
   class << self
+    KNOWN = [:md5, :sha, :sha1, :sha256, :sha384, :sha512]
     # Generate a password-hash suitable for inclusion in an LDAP attribute.
     # Pass a hash type as a symbol (:md5, :sha, :ssha) and a plaintext
     # password. This function will return a hashed representation.
     #
     #--
     # STUB: This is here to fulfill the requirements of an RFC, which
     # one?
-    #
-    # TODO:
-    # * maybe salted-md5
-    # * Should we provide sha1 as a synonym for sha1? I vote no because then
-    #   should you also provide ssha1 for symmetry?
-    #
-    attribute_value = ""
     def generate(type, str)
-      case type
-      when :md5
-         attribute_value = '{MD5}' + Base64.encode64(Digest::MD5.digest(str)).chomp!
-      when :sha
-         attribute_value = '{SHA}' + Base64.encode64(Digest::SHA1.digest(str)).chomp!
-      when :ssha
-         salt = SecureRandom.random_bytes(16)
-         attribute_value = '{SSHA}' + Base64.encode64(Digest::SHA1.digest(str + salt) + salt).chomp!
```

Comment:
Can we keep the same pattern here without introducing a KNOWN constant? Yes, there is more duplication, but it's simpler to reason about in my opinion and more obvious to grep for than hiding the information via lookups and `module_evals`.